test/Microsoft.Azure.Devices.Edge.Test/helpers/ManualProvisioningFixture.cs

```diff
@@ -30,63 +34,61 @@ public ManualProvisioningFixture()
                 this.iotHub);
         }
 
-        [OneTimeSetUp]
-        public async Task ManuallyProvisionEdgeAsync()
+        public async Task ManuallyProvisionEdgeSasAsync(EdgeDevice device, DateTime startTime, CancellationToken token)
         {
-            await Profiler.Run(
-                async () =>
+            IotHubConnectionStringBuilder builder =
+                IotHubConnectionStringBuilder.Create(device.ConnectionString);
+
+            await this.daemon.ConfigureAsync(
+                config =>
                 {
-                    using (var cts = new CancellationTokenSource(Context.Current.SetupTimeout))
-                    {
-                        // NUnit's [Timeout] attribute isn't supported in .NET Standard
-                        // and even if it were, it doesn't run the teardown method when
-                        // a test times out. We need teardown to run, to remove the
-                        // device registration from IoT Hub and stop the daemon. So
-                        // we have our own timeout mechanism.
-                        DateTime startTime = DateTime.Now;
-                        CancellationToken token = cts.Token;
+                    config.SetDeviceConnectionString(device.ConnectionString);
+                    config.Update();
+                    return Task.FromResult((
+                        "with connection string for device '{Identity}'",
+                        new object[] { builder.DeviceId }));
+                },
+                token);
 
-                        EdgeDevice device = await EdgeDevice.GetOrCreateIdentityAsync(
-                            Context.Current.DeviceId,
-                            this.iotHub,
-                            token);
-                        Context.Current.DeleteList.TryAdd(device.Id, device);
+            await this.WaitForConfiguredStatusAsync(device, startTime, token);
+        }
 
-                        IotHubConnectionStringBuilder builder =
-                            IotHubConnectionStringBuilder.Create(device.ConnectionString);
+        public async Task ManuallyProvisionEdgeX509Async(EdgeDevice device, string certPath, string keyPath, DateTime startTime, CancellationToken token)
+        {
+            await this.daemon.ConfigureAsync(
+                config =>
+                {
+                    config.SetDeviceManualX509(device.HubHostname, device.Id, certPath, keyPath);
+                    config.Update();
+                    return Task.FromResult((
+                        "with connection string for device '{Identity}'",
```

Comment:
Change the message to something like `"with X509 certificate for device '{Identity}'"`